pytorch_pfn_extras/handler/_logic.py

```diff
@@ -464,3 +465,198 @@ def eval_step(
         model = models[self.model_name]
         outs = forward(model)(batch)
         return outs
+
+
+@dataclasses.dataclass
+class ClousureModelOutput:
+    outs: Any
+    loss: torch.Tensor
+
+    def __float__(self) -> float:
+        return float(self.loss)
+
+
+class ClousureLogic(BaseLogic):
```

Comment:
```suggestion
class ClousureLogic(Logic):
```

Maybe it is better just to specialize `Logic` so we can avoid repeating all the methods :)
I think with this, only `train_step` becomes necessary.

Reply:
Corrected the relevant code.